core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/chunkedencoding/Chunk.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.chunkedencoding;
+
+import static software.amazon.awssdk.utils.FunctionalUtils.invokeSafely;
+
+import java.io.InputStream;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.http.auth.aws.internal.io.ChunkInputStream;
+import software.amazon.awssdk.utils.SdkAutoCloseable;
+
+/**
+ * An interface which defines a "chunk" of data.
+ */
+@SdkProtectedApi
+public interface Chunk extends SdkAutoCloseable {
```

Comment:
This can be internal, because it's not returned anywhere, and it doesn't seem to make a lot of sense to create directly.